services/web/client/source/class/qxapp/components/workbench/NodeBase.js

```diff
@@ -61,67 +76,160 @@ qx.Class.define("qxapp.components.workbench.NodeBase", {
       apply : "__applyMetadata"
     },
 
-    inputLinkIDs: {
-      check: "Array",
-      init: [],
-      nullable: false
-    },
-
     outputLinkIDs: {
       check: "Array",
       init: [],
       nullable: false
     }
   },
 
+  events: {
+    "LinkDragStart": "qx.event.type.Data",
+    "LinkDragOver": "qx.event.type.Data",
+    "LinkDrop": "qx.event.type.Data",
+    "LinkDragEnd": "qx.event.type.Data",
+    "NodeMoving":  "qx.event.type.Data"
+  },
+
   members: {
     __inputPorts: null,
     __outputPorts: null,
+    __inputPortsUI: null,
+    __outputPortsUI: null,
     __progressLabel: null,
+    __progressBar: null,
+
+    getInputPorts: function() {
+      return this.__inputPorts;
+    },
+
+    getOutputPorts: function() {
+      return this.__outputPorts;
+    },
+
+    // override qx.ui.window.Window "move" event listener
+    _onMovePointerMove: function(e) {
+      this.base(arguments, e);
+      if (e.getPropagationStopped() === true) {
+        this.fireDataEvent("NodeMoving", null);
```

Comment:
how about determining the position here, then you would not have to test if there is a content element and dom node, because the fact that the thing is being moved already implies that there must be a dom node

Reply:
That would require changing the way the points for the connection-link are calculated. I personally prefer this approach.